frontend/catalyst/cuda/__init__.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains a CudaQDevice and the qjit
+entry point.
+"""
+
+from pathlib import Path
+
+import cudaq
+import pennylane as qml
+
+from catalyst.cuda.catalyst_to_cuda_interpreter import interpret
+
+
+def qjit(fn=None, **kwargs):
```

Comment:
A quick question I have: because `qjit` is not re-using/subclassing the Catalyst `QJIT` class, it doesn't have support of features like AOT (based on type signatures), autograph, etc.

Out of scope for this PR, but do you see a straightforward pathway for adding these features in the future?

Reply:
> @erick-xanadu is this the main user entry point for the catalyst QJIT?

No. This is only the entry point for CUDA catalyst QJIT. This is different from the Catalyst QJIT. (A bit confusing, two different functions).

> it doesn't have support of features like AOT (based on type signatures), autograph, etc.

Right now we generate JAXPR every single time the function is called and then evaluate this JAXPR according to the CUDA Quantum interpreter, which executes the quantum instructions in CUDA Quantum. We can evaluate it abstractly in order to generate a new JAXPR and save this JAXPR representation. I think this can be added somewhat easily.